hooks/useCustomStyle.tsx

```diff
@@ -0,0 +1,122 @@
+function getPseudoElementWidth(currentElement: HTMLElement, pseudoName : string) {
+
+    return Number(getComputedStyle(currentElement, pseudoName).width.replace('px', ''))
+}
+
+function isInCodeContainer(currentElement : HTMLElement) {
+    const containerElement = currentElement.closest('.code-container')!
+    return containerElement?.scrollLeft
+}
+
+function getValuesForCSSLeft( currentElement : HTMLElement) {
+
+    const screenWidth = globalThis.window.outerWidth;
+    const mainElement = document.querySelector('main')!
+    const mainLeftPadding = getOneSidePadding(mainElement)
+
+    const allowedContentFullWidth = screenWidth - (mainLeftPadding * 2)
+
+    const elementXPos = currentElement.getBoundingClientRect().x - mainLeftPadding
+    const elementXPosRatio = elementXPos / allowedContentFullWidth
+
+    return {
+        elementXPos,
+        allowedContentFullWidth,
+        elementXPosRatio,
+    }
+}
+
+function computeTwoSlashOverrideStyle({isMounted,
+                                        contentWidth,
+                                        allowedContentFullWidth,
+                                        elementXPos,
+                                      } :
+                                      {isMounted: boolean,
+                                      contentWidth: number ,
+                                      allowedContentFullWidth: number ,
+                                      elementXPos: number ,
+                                      }) {
+
+    return  isMounted
+            ?(
+                contentWidth >= allowedContentFullWidth
+                ? `pre.twoslash data-lsp:hover::before {
+                left: 0px !important;
+                }`
+                : elementXPos + contentWidth >= allowedContentFullWidth
+                ? `pre.twoslash data-lsp:hover::before {
+                left: ${allowedContentFullWidth - contentWidth}px !important;
+                }`
+                : `pre.twoslash data-lsp:hover::before {
+                left: ${elementXPos}px !important;
+                }`
+            )
+            : "";
+}
+
+export function useCustomStyle() {
+
+  const [isMounted, setIsMounted] = useState(false);
+  const [currentStyle, setCurrentStyle] = useState("");
+
+  function rePositionTwoSlashBox(e:Event) {
+
+    const currentElement = e.target as (HTMLElement | null)
+    if( currentElement == null || !isMobile()) return
+    if(isInCodeContainer(currentElement) == null) return
+
+    const {
+            elementXPos,
+            allowedContentFullWidth,
+        } =  getValuesForCSSLeft(currentElement)
+
+    const contentWidth = getPseudoElementWidth(currentElement, ':before');
+
+    setCurrentStyle(
+        computeTwoSlashOverrideStyle({
+            isMounted,
+            contentWidth,
+            allowedContentFullWidth,
+            elementXPos,
+        })
+    )
+  }
+
+  useEffect(() => {
+      setIsMounted(true);
+      return () => setIsMounted(false);
+  },[])
```

Comment:
the flow here is as follows:
```
mount 
-> effect (setsMount) [trigger rerender] 
-> effect (add listeners -- not mounted )
-> rerender
-> effect {deps changed} (add listeners -- is mounted)
```
not really sure what is the point..

`useEffect` _runs only in the client side_ and not on SSR or SSG, meaning that it is safe to access the `window` object inside an effect. This mount check is redundant

Reply:
change the name "isMounted" to "isClientReady"

the logic in useEffect no longer depends on this